contracts/zero-ex/contracts/src/features/OtcOrdersFeature.sol

```diff
@@ -0,0 +1,443 @@
+    string public constant override FEATURE_NAME = "OtcOrders";
+    /// @dev Version of this feature.
+    uint256 public immutable override FEATURE_VERSION = _encodeVersion(1, 0, 0);
+    /// @dev The WETH token contract.
+    IEtherTokenV06 private immutable WETH;
+
+    constructor(address zeroExAddress, IEtherTokenV06 weth)
+        public
+        FixinEIP712(zeroExAddress)
+    {
+        WETH = weth;
+    }
+
+    /// @dev Initialize and register this feature.
+    ///      Should be delegatecalled by `Migrate.migrate()`.
+    /// @return success `LibMigrate.SUCCESS` on success.
+    function migrate()
+        external
+        returns (bytes4 success)
+    {
+        _registerFeatureFunction(this.fillOtcOrder.selector);
+        _registerFeatureFunction(this.fillTakerSignedOtcOrder.selector);
+        _registerFeatureFunction(this.getOtcOrderInfo.selector);
+        _registerFeatureFunction(this.getOtcOrderHash.selector);
+        _registerFeatureFunction(this.minTxOriginNonce.selector);
+        return LibMigrate.MIGRATE_SUCCESS;
+    }
+
+    /// @dev Fill an OTC order for up to `takerTokenFillAmount` taker tokens.
+    /// @param order The OTC order.
+    /// @param makerSignature The order signature from the maker.
+    /// @param takerTokenFillAmount Maximum taker token amount to fill this
+    ///        order with.
+    /// @param unwrapWeth Whether or not to unwrap bought WETH into ETH
+    ///        before transferring it to the taker. Should be set to false
+    ///        if the maker token is not WETH.
+    /// @return takerTokenFilledAmount How much taker token was filled.
+    /// @return makerTokenFilledAmount How much maker token was filled.
+    function fillOtcOrder(
+        LibNativeOrder.OtcOrder memory order,
+        LibSignature.Signature memory makerSignature,
+        uint128 takerTokenFillAmount,
+        bool unwrapWeth
+    )
+        public
+        override
+        returns (uint128 takerTokenFilledAmount, uint128 makerTokenFilledAmount)
+    {
+        if (!_isSenderValidTaker(order.taker)) {
+            bytes32 orderHash = getOtcOrderHash(order);
+            LibNativeOrdersRichErrors.OrderNotFillableByTakerError(
+                orderHash,
+                msg.sender,
+                order.taker
+            ).rrevert();
+        }
+        LibSignature.Signature memory nullSignature;
+        return _fillOtcOrderPrivate(
+            order,
+            makerSignature,
+            nullSignature,
+            takerTokenFillAmount,
+            unwrapWeth ? WethOptions.UnwrapWeth : WethOptions.LeaveAsWeth
+        );
+    }
+
+    /// @dev Fill an OTC order whose taker token is WETH for up
+    ///      to `msg.value`.
+    /// @param order The OTC order.
+    /// @param makerSignature The order signature from the maker.
+    /// @return takerTokenFilledAmount How much taker token was filled.
+    /// @return makerTokenFilledAmount How much maker token was filled.
+    function fillOtcOrderWithEth(
```

Comment:
When would we ever use this function?
- In RFQM, the taker does not have opportunity to send ETH because they are not submitting the trade
- In RFQT, we usually use the VIP+Fallback route when interacting with `fillOtcOrder`. Is that route `payable`?, also, If the taker was to send ETH with the VIP fallback route and the order reverts and falls back to Uni, what would happen to the ETH?

Reply:
If someone is filling a RFQt order with no fallback (e.g. for OTC) and they want to use ETH instead of WETH